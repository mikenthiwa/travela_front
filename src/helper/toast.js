import toast from 'toastr';
import cancelBtn from '../images/toast-cancel.svg';

toast.options = {
  progressBar: false,
  closeButton: true,
  preventDuplicates: true,
  positionClass: 'toast-top-center',
  closeHtml: `
    <button class="custom-close-btn">
      <img alt="close btn"/>
    </button>
  `,
};

export const successMessage = message =>  toast.success(message);
export const errorMessage = message =>  toast.error(message);


