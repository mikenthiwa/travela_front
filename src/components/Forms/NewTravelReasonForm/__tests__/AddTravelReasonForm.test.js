import React from 'react';
import toast from 'toastr';
import moxios from 'moxios';
import AddTravelReason from '../index';


const props = {
  errors: {},
  createNewTravelReason: jest.fn(),
  closeModal: jest.fn(),
  travelReason: {},
  isLoading: false
};

toast.error = jest.fn();
const event = {
  preventDefault: jest.fn(),
};

describe('<AddTravelReason />', () => {
  const wrapper = mount(<AddTravelReason {...props} />);
  beforeEach(() => {
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
    toast.error.mockReset();
  });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handles submit', () => {
    wrapper.find('.new-request').simulate('submit', event);
    moxios.wait(() => {
      expect(props.createTravelReason).toHaveBeenCalled();
    });
  });

  it('should validate title over 18 characters', () => {
    const wrapped = shallow(<AddTravelReason {...props} />);

    const inputWrapper = wrapped.instance();

    inputWrapper.state = {
      ...inputWrapper.state,
      values: {
        title: '12345663644737382833478'
      }
    };

    inputWrapper.validateLength('title');
    moxios.wait(() => {
      expect(inputWrapper.validateField('title')).toHaveBeenCalled();
    });
  });

  it('should validate title less than 18 characters', () => {
    const wrapped = shallow(<AddTravelReason {...props} />);

    const inputWrapper = wrapped.instance();

    inputWrapper.state = {
      ...inputWrapper.state,
      values: {
        title: '12345678'
      }
    };

    inputWrapper.validateLength('title');
    moxios.wait(() => {
      expect(inputWrapper.validateField('title')).toHaveBeenCalled();
    });
  });

  it('should validate title ==== 0', () => {
    const wrapped = shallow(<AddTravelReason {...props} />);

    const inputWrapper = wrapped.instance();

    inputWrapper.state = {
      ...inputWrapper.state,
      values: {
        title: ''
      }
    };

    inputWrapper.validateLength('title');
    moxios.wait(() => {
      expect(inputWrapper.validateField('title')).toHaveBeenCalled();
    });
  });

  it('should validate description', () => {
    const wrapped = shallow(<AddTravelReason {...props} />);

    const inputWrapper = wrapped.instance();

    inputWrapper.state = {
      ...inputWrapper.state,
      values: {
        description: '1hjkhdfs;iugfiufgdsfigfjgsigfiujfmghfiugugfijdsgfiesgf7wg7e234567890poiuyttrrewqasddffghjkkl09987642135890lopikjgfd12456789876543213456789087654321234567'
      }
    };

    inputWrapper.validateLength('description');
    moxios.wait(() => {
      expect(inputWrapper.validateField('description')).toHaveBeenCalled();
    });
  });

  it('should validate description when over 140 characters', () => {
    const wrapped = shallow(<AddTravelReason {...props} />);

    const inputWrapper = wrapped.instance();

    inputWrapper.validateLength('description');
    moxios.wait(() => {
      expect(inputWrapper.validateField('description')).toHaveBeenCalled();
    });
  });

  it('should handle default validation case', () => {
    const wrapped = shallow(<AddTravelReason {...props} />);

    const inputWrapper = wrapped.instance();

    inputWrapper.validateLength('');
    moxios.wait(() => {
      expect(inputWrapper.validateField('')).toHaveBeenCalled();
    });
  });

  it('closes modal on cancel', () => {
    expect(wrapper.find('#cancel').length).toEqual(1);
    wrapper.find('#cancel').simulate('click', event);
    expect(props.closeModal).toHaveBeenCalled();
  });

});
