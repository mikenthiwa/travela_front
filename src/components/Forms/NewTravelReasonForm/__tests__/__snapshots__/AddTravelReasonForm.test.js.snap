// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTravelReason /> renders without crashing 1`] = `
<form
  className="new-request"
  key=".0"
  onSubmit={[Function]}
>
  <fieldset
    className="add-checklist"
  >
    <div
      className="input-group"
    >
      <div
        className="form-input  "
      >
        <label
          htmlFor="title"
        >
          Title
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
             * 
          </span>
        </label>
        <input
          autoComplete="off"
          className=" "
          label="Title"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <span
          className="error"
        />
      </div>
      <div
        className="form-input  "
      >
        <label
          htmlFor="description"
        >
          Description
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
             * 
          </span>
        </label>
        <div
          className="form-input__textarea "
        >
          <textarea
            autoComplete="off"
            className=" textarea-box "
            data-gramm_editor="false"
            label="Description"
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            rows="10"
            type="textarea"
            value=""
          />
        </div>
        <span
          className="error"
        />
      </div>
    </div>
  </fieldset>
  <fieldset>
    <div
      className="submit-area delete-checklist-item__right"
    >
      <button
        className="bg-btn bg-btn--inactive"
        id="cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="bg-btn bg-btn--active"
        disabled={true}
        type="submit"
      >
        <undefined />
      </button>
    </div>
  </fieldset>
</form>
`;
